# Upstream status: won't go into Linus' tree like this but is in the
# linux-mips tree.  The drivers/char serial drivers need to be converted to
# real serial drivers in drivers/serial

# Author: Martin Michlmayr <tbm@cyrius.com>, mostly taken from the
# linux-mips tree


--- b/drivers/char/Kconfig~	2006-05-25 12:51:58.000000000 +0200
+++ b/drivers/char/Kconfig	2006-05-25 12:53:03.000000000 +0200
@@ -362,6 +362,41 @@
 	bool "Console on BCM1xxx DUART"
 	depends on SIBYTE_SB1250_DUART
 
+config SERIAL_DEC
+	bool "DECstation serial support"
+	depends on MACH_DECSTATION
+	default y
+	help
+	  This selects whether you want to be asked about drivers for
+	  DECstation serial ports.
+
+	  Note that the answer to this question won't directly affect the
+	  kernel: saying N will just cause the configurator to skip all
+	  the questions about DECstation serial ports.
+
+	  If unsure, say Y.
+
+config SERIAL_DEC_CONSOLE
+	bool "Support for console on a DECstation serial port"
+	depends on SERIAL_DEC
+	default y
+	help
+	  If you say Y here, it will be possible to use a serial port as the
+	  system console (the system console is the device which receives all
+	  kernel messages and warnings and which allows logins in single user
+	  mode).  Note that the firmware uses ttyS0 as the serial console on
+	  the Maxine and ttyS2 on the others.
+
+	  If unsure, say Y.
+
+config ZS
+	bool "Z85C30 Serial Support"
+	depends on SERIAL_DEC
+	default y
+	help
+	  Documentation on the Zilog 85C350 serial communications controller
+	  is downloadable at <http://www.zilog.com/pdfs/serial/z85c30.pdf>.
+
 config QTRONIX_KEYBOARD
 	bool "Enable Qtronix 990P Keyboard Support"
 	depends on IT8712
--- b/drivers/char/Makefile~	2006-05-25 12:51:52.000000000 +0200
+++ b/drivers/char/Makefile	2006-05-25 12:52:27.000000000 +0200
@@ -50,6 +50,7 @@
 obj-$(CONFIG_VIOTAPE)		+= viotape.o
 obj-$(CONFIG_HVCS)		+= hvcs.o
 obj-$(CONFIG_SGI_MBCS)		+= mbcs.o
+obj-$(CONFIG_SERIAL_DEC)	+= decserial.o
 
 obj-$(CONFIG_PRINTER) += lp.o
 obj-$(CONFIG_TIPAR) += tipar.o
--- a/drivers/char/decserial.c
+++ b/drivers/char/decserial.c
@@ -24,17 +24,17 @@
 extern int zs_init(void);
 #endif
 
-#ifdef CONFIG_DZ
+#ifdef CONFIG_SERIAL_DZ
 extern int dz_init(void);
 #endif
 
-#ifdef CONFIG_SERIAL_CONSOLE
+#ifdef CONFIG_SERIAL_CORE_CONSOLE
 
 #ifdef CONFIG_ZS
 extern void zs_serial_console_init(void);
 #endif
 
-#ifdef CONFIG_DZ
+#ifdef CONFIG_SERIAL_DZ
 extern void dz_serial_console_init(void);
 #endif
 
@@ -43,12 +43,12 @@ extern void dz_serial_console_init(void)
 /* rs_init - starts up the serial interface -
    handle normal case of starting up the serial interface */
 
-#ifdef CONFIG_SERIAL
+#ifdef CONFIG_SERIAL_CORE
 
 int __init rs_init(void)
 {
 
-#if defined(CONFIG_ZS) && defined(CONFIG_DZ)
+#if defined(CONFIG_ZS) && defined(CONFIG_SERIAL_DZ)
     if (IOASIC)
 	return zs_init();
     else
@@ -59,7 +59,7 @@ int __init rs_init(void)
     return zs_init();
 #endif
 
-#ifdef CONFIG_DZ
+#ifdef CONFIG_SERIAL_DZ
     return dz_init();
 #endif
 
@@ -70,14 +70,14 @@ __initcall(rs_init);
 
 #endif
 
-#ifdef CONFIG_SERIAL_CONSOLE
+#ifdef CONFIG_SERIAL_CORE_CONSOLE
 
 /* serial_console_init handles the special case of starting
  *   up the console on the serial port
  */
 static int __init decserial_console_init(void)
 {
-#if defined(CONFIG_ZS) && defined(CONFIG_DZ)
+#if defined(CONFIG_ZS) && defined(CONFIG_SERIAL_DZ)
     if (IOASIC)
 	zs_serial_console_init();
     else
@@ -88,7 +88,7 @@ static int __init decserial_console_init
     zs_serial_console_init();
 #endif
 
-#ifdef CONFIG_DZ
+#ifdef CONFIG_SERIAL_DZ
     dz_serial_console_init();
 #endif
 
